libxrdp/xrdp_sec.c

```diff
@@ -2302,159 +2302,235 @@ xrdp_sec_process_mcs_data_channels(struct xrdp_sec *self, struct stream *s)
     return 0;
 }
 
-/*****************************************************************************/
-/* Process a [MS-RDPBCGR] TS_UD_CS_MONITOR message.
-   reads the client monitors data */
-static int
-xrdp_sec_process_mcs_data_monitors(struct xrdp_sec *self, struct stream *s)
+int
+process_monitor_stream(struct stream *s, struct display_size_description *description, int full_parameters)
 {
-    int index;
-    int monitorCount;
-    int flags;
-    int x1;
-    int y1;
-    int x2;
-    int y2;
-    int got_primary;
-    struct xrdp_client_info *client_info = (struct xrdp_client_info *)NULL;
-
-    client_info = &(self->rdp_layer->client_info);
-
-    /* this is an option set in xrdp.ini */
-    if (client_info->multimon != 1) /* are multi-monitors allowed ? */
-    {
-        LOG(LOG_LEVEL_INFO, "Multi-monitor is disabled by server config");
-        return 0;
-    }
-    if (!s_check_rem_and_log(s, 8, "Parsing [MS-RDPBCGR] TS_UD_CS_MONITOR"))
+    LOG_DEVEL(LOG_LEVEL_TRACE, "process_monitor_stream:");
+    if (description == NULL)
     {
+        LOG_DEVEL(LOG_LEVEL_DEBUG, "\tprocess_monitor_stream: description was null. Valid pointer to allocated description expected.");
```

Comment:
Also, I agree this should be LOG_LEVEL_ERROR

Reply:
Another also - please split the string into multiple string literals `"..." "..."` and reformat to 80 chars (or a little bit over) max line width as per the coding standard. There are quite a few other instances of this further below too. This bit of shell magic picks out the culprits:-

```
grep -n '' libxrdp/xrdp_sec.c | sed -ne '2307,2505p' | grep -E ':.{80}'
```

I appreciate that other bits of this file are also breaking the standard here, but for new code I think we should try to avoid this where possible.